inference-engine/src/mkldnn_plugin/nodes/mkldnn_one_hot_node.cpp

```diff
@@ -56,32 +51,63 @@ MKLDNNOneHotNode::MKLDNNOneHotNode(const std::shared_ptr<ngraph::Node>& op, cons
     errorPrefix = "OneHot layer with name '" + op->get_friendly_name() + "'";
     const auto oneHot = std::dynamic_pointer_cast<const ngraph::opset1::OneHot>(op);
     const auto depthNode = std::dynamic_pointer_cast<const ngraph::opset1::Constant>(oneHot->get_input_node_shared_ptr(DEPTH_ID));
-    const auto onValueNode = std::dynamic_pointer_cast<const ngraph::opset1::Constant>(oneHot->get_input_node_shared_ptr(ON_VALUE_ID));
-    const auto offValueNode = std::dynamic_pointer_cast<const ngraph::opset1::Constant>(oneHot->get_input_node_shared_ptr(OFF_VALUEAXES_ID));
     depth = depthNode->cast_vector<uint32_t>()[0];
     axis = oneHot->get_axis();
-    src_dims = oneHot->get_input_shape(INDICES_ID);
-    if (ngraph::is_scalar(src_dims)) {
-        src_dims = SizeVector{1};
+
+    VectorDims srcDims = inputShapes[INDICES_ID].getDims();
```

Comment:
why you leave check that check must be `Constant` in `isSupportedOperation`?

also remove `const auto depthNode = std::dynamic_pointer_cast<const ngraph::opset1::Constant>(oneHot->get_input_node_shared_ptr(DEPTH_ID));` from ctor

better to use
```suggestion
    VectorDims srcDims = getInputShapeAtPort(INDICES_ID).getDims();
```